bliss/simulator/prior.py

```diff
@@ -199,103 +224,33 @@ def multiflux(self, rcf_ratios, flux_base) -> Tensor:
-                )
-                assert Path(po_path).exists(), msg
-                po_fits = fits.getdata(po_path)
-
-                # retrieve object-specific information for ratio computing
-                objc_type = column_to_tensor(po_fits, "objc_type").numpy()
-                thing_id = column_to_tensor(po_fits, "thing_id").numpy()
-
-                # mask fluxes based on object identity & validity
-                galaxy_bools = (objc_type == 3) & (thing_id != -1)
-                star_bools = (objc_type == 6) & (thing_id != -1)
-                star_fluxes = column_to_tensor(po_fits, "psfflux") * star_bools.reshape(-1, 1)
-                gal_fluxes = column_to_tensor(po_fits, "cmodelflux") * galaxy_bools.reshape(-1, 1)
-                fluxes = star_fluxes + gal_fluxes
-
-                # containers for light source ratios in current field
-                star_fluxes_rest = []
-                gal_fluxes_rest = []
-
-                # SDSS object for current field
-                sdss_obj = sdss[i]
-
-                for obj, _ in enumerate(objc_type):
-                    if thing_id[obj] != -1 and torch.all(fluxes[obj] > 0):
-                        ratios = self._compute_flux_ratio(fluxes[obj], sdss_obj)  # noqa: WPS220
-                        if objc_type[obj] == 6:  # noqa: WPS220
-                            star_fluxes_rest.append(ratios)  # noqa: WPS220
-                        elif objc_type[obj] == 3:  # noqa: WPS220
-                            gal_fluxes_rest.append(ratios)  # noqa: WPS220
-
-                if star_fluxes_rest:
-                    rcf_stars_rest[(run, camcol, field)] = star_fluxes_rest
-                if gal_fluxes_rest:
-                    rcf_gals_rest[(run, camcol, field)] = gal_fluxes_rest
-        return rcf_stars_rest, rcf_gals_rest
-
-    @staticmethod
-    def _draw_truncated_pareto(alpha, min_x, max_x, n_samples) -> Tensor:
-        # draw pareto conditioned on being less than f_max
-        u_max = 1 - (min_x / max_x) ** alpha
-        uniform_samples = torch.rand(n_samples) * u_max
-        return min_x / (1.0 - uniform_samples) ** (1 / alpha)
+    def _flux_ratios_against_b(self, survey_data_dir, image_ids, items, b) -> Tuple[dict, dict]:
+        """Sample and compute all star, galaxy fluxes relative to `b`-band based on real image data.
 
-    def _compute_flux_ratio(self, obj_fluxes, sdss_obj):
-        """Query SDSS frames to get flux ratios in units of electron count.
+        Instead of pareto-sampling fluxes for each band, we pareto-sample `b`-band flux values,
+        using prior `b`-band star/galaxy flux parameters, then apply other-band-to-`b`-band flux
+        ratios to get other-band flux values. This is primarily because Pareto distribution is
+        multimodal, and so it is not always the case that, e.g.,
+            flux_r ~ Pareto, flux_g ~ Pareto => flux_r | flux_g ~ Pareto.
```

Comment:
This isn't entirely accurate - Pareto is unimodal ([PDF](https://s3.amazonaws.com/static.wd7.us/d/d9/Pareto_distributionPDF.png)). I meant that the conditional distribution is multimodal, not the marginal. It's probably fine to leave this for now since I think a lot of this code is going to be replaced with Sawan's GMM ratio model, but we should update it then to give more details about the color distribution @sawanp813